lib/core/src/Cardano/Wallet/Transaction.hs

```diff
@@ -68,9 +72,21 @@ data TransactionLayer t k = TransactionLayer
         -- This expects as a first argument a mean to compute or lookup private
         -- key corresponding to a particular address.
 
+    , mkDelegationCertificates
+        :: W.DelegationAction
+            -- Pool Id to which we're planning to delegate
+        -> (XPrv, Passphrase "encryption")
+            -- Reward account
+        -> CoinSelection
```

Comment:
Our current logic is a bit convoluted / backward mostly because of the Jormungandr support. In principle, we construct the coin selection from the certificates, not the other way around. We can indeed _infer_ what certificates were assumed for the coin-selection but that information is already solely available from `DelegationAction`. 

RegisterKeyAndJoin -> a stake key certificate + stake delegation certificate
Join -> a stake delegation certificate
Quit -> a stake key de-registration certificate

Reply:
Yeah, I saw that and I was feeling a bit odd about it. Also because we have two `DelegationAction` types now... one just `Join | Quit` (for the API input) and the other what you wrote above.

Reply:
It'd be okay to define use `ApiT DelegationAction` in the API, and have the JSON instance fails on parsing `RegisterKeyAndJoin` so that only `Join` and `Quit` are exposed in the JSON interface.